### PR TITLE
Add location to AST::Visibility

### LINKED ISSUE #1183: Add location to `AST::Visibility`
And use it when lowering instances to `HIR::Visibility`s in `translate_visibility` (gcc/rust/hir/rust-ast-lower.cc)

### PR DESCRIPTION
**Pull request by dme2**
Hi. I've added location to AST::Visibility, but it's possible I may have missed some steps here.

Fixes #1183.

### CODE REVIEW

**Review by CohenArthur — COMMENTED** [review]
Thank you for working on this! This is in a great state minus one tiny nitpick :)

File: gcc/rust/ast/rust-item.h [hunk 1]
```diff
@@ -698,7 +705,7 @@ struct Visibility
   // Creates a public visibility with a given path or whatever.
   static Visibility create_in_path (SimplePath in_path)
   {
-    return Visibility (PUB_IN_PATH, std::move (in_path));
+    return Visibility (PUB_IN_PATH, std::move (in_path), in_path.get_locus ());
```
[the remark this answers is not in the document]
Comment (dme2) [reply gcc/rust/ast/rust-item.h:708]: Thank you for the feedback. I think I'll go ahead and work on adding the fixes to this PR.

[the remark this answers is not in the document]
Comment (CohenArthur) [reply gcc/rust/ast/rust-item.h:708]: > Thank you for the feedback. I think I'll go ahead and work on adding the fixes to this PR.

Great! Feel free to ask any questions or anything that you might need to go further

**Review by CohenArthur — APPROVED** [review]
This looks great :) Is there any reason the PR is still marked as draft?

**Comment by dme2** [comment]
Nope, no reason :)

**Comment by CohenArthur** [comment]
Nice! Thank you for the work @dme2 

bors r+

**Pull request merged**